core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -94,6 +101,86 @@ public final void setPort(final int port) {
         }
     }
 
+    /**
+     * Gets the name or literal IP address of the proxy host that the client is configured with.
+     *
+     * @return The host name.
+     */
+    public final String getProxyHost() {
+        return proxyHost;
+    }
+
+    /**
+     * Sets the name or literal IP address of the proxy host that the client should use to connect to the an AMQP host
+     *
+     * @param proxyHost The host name or IP address of the proxy
+     * @throws NullPointerException if proxyHost is {@code null}.
+     */
+    public final void setProxyHost(final String proxyHost) {
+        this.proxyHost = Objects.requireNonNull(proxyHost);
+    }
+
+    /**
+     * Gets the TCP port of the proxy server that this client is configured to use.
+     *
+     * @return The port number.
+     */
+    public final int getProxyPort() {
+        return proxyPort;
+    }
+
+    /**
+     * Sets the TCP port of the proxy server that this client should use.
+     *
+     * @param proxyPort The port number of the proxy
+     * @throws IllegalArgumentException if port &lt; 1000 or port &gt; 65535.
+     */
+    public final void setProxyPort(final int proxyPort) {
+        if (isValidPort(proxyPort)) {
```

Comment:
:+1:  for using the helper class here.
But did you see that this results in "proxyPort == 0" being a valid value? Different what you had before.
And: not matching with the javadoc above (<= 1000 or > 65535).
Please adopt, thanks!

Reply:
I see now that in the helper class PortConfigurationHelper.MIN_PORT_NO is set to 0. This means, that setPort() method was also implemented wrongly and the javadoc is not correct. I would prefer to set  PortConfigurationHelper.MIN_PORT_NO to 1000 instead of 0, WDYT?

Reply:
1.) why should a configuration like "HTTP adapter set to port 80" be always wrong?
2.) Why `1000` as a limit? I would rather set a limit to `1024`, since on most systems you need root access to open a port less than `1024`.

I would not adopt the helper class, and only add a check in your code to disallow the `0`, as you already had before.

Reply:
Ok, I will then add the same check also to setPort so that it is consistent.